java/integTest/src/test/java/glide/cluster/CommandTests.java

```diff
@@ -655,11 +655,27 @@ public void dbsize() {
 
         // test dbsize with routing - flush the database first to ensure the set() call is directed to a
         // node with 0 keys.
-        assertEquals(OK, clusterClient.flushall().get());
+        assertEquals(OK, clusterClient.flushdb().get());
         assertEquals(0L, clusterClient.dbsize().get());
+
         String key = UUID.randomUUID().toString();
+        SingleNodeRoute route = new SlotKeyRoute(key, PRIMARY);
+
+        // add a key, measure DB size, flush DB and measure again - with all arg combinations
+        assertEquals(OK, clusterClient.set(key, "foo").get());
+        assertEquals(1L, clusterClient.dbsize(route).get());
+        assertEquals(OK, clusterClient.flushdb(ASYNC).get());
+        assertEquals(0L, clusterClient.dbsize().get());
```

Comment:
Could this potentially be flaky since the flushdb above is ASYNC? In other words, is there a chance the ASYNC flushdb call has not completed yet so the dbsize would return 1?